includes/sanitizers/class-amp-nav-menu-toggle-sanitizer.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Class AMP_Nav_Menu_Toggle_Sanitizer
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Nav_Menu_Toggle_Sanitizer
+ *
+ * Handles state for navigation menu toggles, based on theme support.
+ *
+ * @since 1.1.0
+ */
+class AMP_Nav_Menu_Toggle_Sanitizer extends AMP_Base_Sanitizer {
```

Comment:
What about making this instead a `AMP_Nav_Menu_Sanitizer`? Ideally this would handle not just the hamburger menu but also the injection of the submenu expand/collapse buttons. In other words, the logic in the `add_nav_sub_menu_buttons` method of the core theme sanitizer should be included here as well, in addition to the logic from the `add_nav_menu_toggle` class.

Eventually I would hope that AMP could actually provide a built-in nav menu component which could handle all of this as part of the framework instead of requiring each site to copy the same logic over and over again. When/if that happens, then much of this class could be removed in favor of just injecting the AMP component. The learnings from this sanitizer could also directly inform the creation of such a nav menu AMP component as well.

Reply:
Also, perhaps the `amp-sidebar` and `amp-accordion` are already the blessed AMP components for doing what WordPress does in a different way and there isn't anything else that AMP needs to add.

Reply:
> What about making this instead a `AMP_Nav_Menu_Sanitizer`?

Sounds good to me. Is it okay if I rename it to that, and we add the extra submenu functionality in a follow-up PR? It wouldn't require any breaking change anyway, would only be an addition.

> Eventually I would hope that AMP could actually provide a built-in nav menu component 

Agreed, I'd be happy to get involved.

> Also, perhaps the `amp-sidebar` and `amp-accordion` are already the blessed AMP components

Possibly, but I wouldn't want to enable one of these by default as replacement for the navigation menu as it would modify behavior a bit. A future enhancement for using `amp-sidebar` instead of the current nav menu logic via opt-in (by theme support) sounds certainly interesting.

Reply:
> Is it okay if I rename it to that, and we add the extra submenu functionality in a follow-up PR? It wouldn't require any breaking change anyway, would only be an addition.

If you must, but this PR seems like the logical place to extract all the menu-specific code from the core theme sanitizer. Why not do it together?

Reply:
> If you must, but this PR seems like the logical place to extract all the menu-specific code from the core theme sanitizer.

Alright, I'll do it here. Revisiting this though, I think we should use a separate class for the submenu toggle sanitizer, mainly because of how theme support should be added to these two things. They are not directly related, so I think it should be possible to add theme support individually. As in:

```php
add_theme_support(
    'amp',
    array(
        'nav_menu_toggle'           => array( ... ),
        'nav_menu_dropdown_toggles' => array( ... ),
    )
);
```

instead of just having one generic `nav_menu` key and cramping everything in there. The separation makes it more obvious what you're adding support for. Then each of the two arrays is passed to the respective sanitizer class.